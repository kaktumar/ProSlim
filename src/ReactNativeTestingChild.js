import React, {
  Component,
  PropTypes
} from 'react';
import {
  StyleSheet,
  View,
  Text,
  TouchableHighlight
} from 'react-native';

import { capitalizeWords } from './ReactNativeTestingUtils'

export default class ReactNativeTestingChild extends Component {
  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>
          Capitalized string:
        </Text>
        <Text style={styles.string}>
          {this._getCapitalizedString()}
        </Text>
        <TouchableHighlight onPress={this.props.onClear}>
          <Text style={styles.button}>
            Clear text
          </Text>
        </TouchableHighlight>
      </View>
    );
  }

  _getCapitalizedString () {
    const { text } = this.props;
    if (text) {
      return capitalizeWords(text);
    }

    return 'You must input something!';
  }
};

ReactNativeTestingChild.propTypes = {
  text: PropTypes.string.isRequired,
  onClear: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
  container: {
    marginTop: 20,
    alignItems: 'center'
  },
  title: {
    fontSize: 20,
    marginBottom: 20
  },
  string: {
    fontSize: 18,
    padding: 20
  },
  button: {
    marginTop: 30,
    fontSize: 30,
    color: '#007aff'
  }
});