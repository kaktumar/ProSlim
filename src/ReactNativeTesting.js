import React, { Component } from 'react';
import {
  StyleSheet,
  Dimensions,
  View,
  Text,
  TextInput,
  Image
} from 'react-native';
import Share from 'react-native-share';
import ReactNativeTestingChild from './ReactNativeTestingChild';

export default class ReactNativeTesting extends Component {
  constructor () {
    super();
    this.handleTextChange = this.handleTextChange.bind(this);
    this.clearText = this.clearText.bind(this);
    this.state = {
      text: ''
    };
  }

  render () {
    const { text } = this.state;
    return (
      <View style={styles.container}>
          <Image style={styles.logo} source={require('./react.png')} />
          <Text style={styles.title}>
              Welcome to React Native testing demo app
          </Text>
          <TextInput
              style={styles.input}
              placeholder={"write something"}
              onChangeText={this.handleTextChange}
              value={text}
          />
          <ReactNativeTestingChild text={text} onClear={this.clearText} />
      </View>
    );
  }

  handleTextChange (text) {
    this._setText(text);
  }

  clearText () {
    this._setText('');
  }

  _setText (text) {
    this.setState({text});
  }
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  title: {
    fontSize: 26,
    padding: 20,
    textAlign: 'center'
  },
  logo: {
    width: 50,
    height: 50
  },
  input: {
    width: (Dimensions.get('window').width - 20),
    height: 60,
    borderColor: 'gray',
    borderWidth: 1,
    borderRadius: 5,
    padding: 10,
    alignSelf: 'center'
  }
});
