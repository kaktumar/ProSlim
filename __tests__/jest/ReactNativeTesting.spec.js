jest.unmock('../../src/ReactNativeTesting');

import '../testutils/jest';
import ReactNativeTesting from '../../src/ReactNativeTesting';
import ReactNativeTestingChild from '../../src/ReactNativeTestingChild';

describe('ReactNativeTesting', () => {
  let component;
  let textInput;
  const defaultState = {text: ''};

  test('renders correctly', () => {
    const snapshot = renderer.create(<ReactNativeTesting />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  beforeEach(() => {
    component = shallow(<ReactNativeTesting />);
    textInput = component.find('TextInput');
  });

  it('has default state', () => {
    expect(component.state()).toEqual(defaultState);
  });

  it('renders welcome text', () => {
    const expectedText = 'Welcome to React Native testing demo app';
    const text = component.find('Text').children().text();
    expect(text).toEqual(expectedText);
  });

  it('renders input field with placeholder', () => {
    const expectedPlaceholder = 'write something';
    expect(textInput.length).toBe(1);
    expect(textInput.props().placeholder).toEqual(expectedPlaceholder);
  });

  describe('when text changes', () => {
    const newTextValue = 'random string';
    beforeEach(() => {
      textInput.simulate('changeText', newTextValue);
    });

    it('updates component state', () => {
      expect(component.state().text).toEqual(newTextValue);
    });

    it('passes text from state to ReactNativeTestingChild', () => {
      const childComponent = component.find(ReactNativeTestingChild);
      expect(childComponent.props().text).toEqual(newTextValue)
    });

    describe('when clearText callback is called', () => {
      beforeEach(() => {
        const childComponent = component.find(ReactNativeTestingChild);
        childComponent.simulate('clear');
      });

      it('resets state', () => {
        expect(component.state()).toEqual(defaultState);
      });
    });
  });
});