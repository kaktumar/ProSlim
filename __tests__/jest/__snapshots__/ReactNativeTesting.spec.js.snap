// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactNativeTesting renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 26,
        "padding": 20,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native testing demo app
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="write something"
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 60,
        "padding": 10,
        "width": 730,
      }
    }
    value=""
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "marginBottom": 20,
        }
      }
    >
      Capitalized string:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "padding": 20,
        }
      }
    >
      You must input something!
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#007aff",
            "fontSize": 30,
            "marginTop": 30,
          }
        }
      >
        Clear text
      </Text>
    </View>
  </View>
</View>
`;
